app/views/comments/_edit.html.erb

```diff
@@ -12,119 +12,48 @@
+            <%= raw translation('comments._edit.drag_and_drop') %>
             </span>
-          </label>
-        </span>
-      </p>
+            <label id="c<%= comment.id%>input_label" for="c<%= comment.id%>input">
+              <input id="c<%= comment.id%>input" type="file" name="image[photo]" style="display:none;" />
+              <a class="hidden-xs"><%= translation('comments._edit.choose_one') %></a>
+              <span class="visible-xs">
+                <i class="fa fa-upload"></i>
+                <a><%= translation('comments._edit.upload_image') %></a>
+              </span>
+            </label>
+          </span>
+        </p>
+        </div>
       </div>
-    </div>
-      <script type="application/javascript">
-        function setInit(id) {
-          const textArea = 'c'+id+'text';
-          const preview = 'c'+id+'preview';
-          $E.setState(textArea, preview);
-        }
-
-        $('#c<%= comment.id%>div').on('dragover',function(e) {
-          e.preventDefault();
-          $('#c<%= comment.id%>div').addClass('hover');
-        });
-
-        $('#c<%= comment.id%>div').on('dragout',function(e) {
-          $('#c<%= comment.id%>div').removeClass('hover');
-        });
-
-        $('#c<%= comment.id%>div').on('drop',function(e) {
-          e.preventDefault();
-          $D.selected = $(e.target).closest('div.comment-form-wrapper').eq(0);
-          setInit(<%= comment.id %>);
-        });
-
-        $('#c<%= comment.id%>div').fileupload({
-          url: "/images",
-          paramName: "image[photo]",
-          dropZone: $('#c<%= comment.id%>div'),
-          dataType: 'json',
-          formData: {
-            'uid':<%= current_user.uid %>,
-            'nid':<%= comment.uid %>
-          },
-          start: function(e) {
-            $('#c<%= comment.id%>progress').show()
-            $('#c<%= comment.id%>uploading').show()
-            $('#c<%= comment.id%>prompt').hide()
-            $('#c<%= comment.id%>div').removeClass('hover');
-          },
-          done: function (e, data) {
-            $('#c<%= comment.id%>progress').hide()
-            $('#c<%= comment.id%>progress #c<%= comment.id%>progress-bar').css('width', 0);
-            $('#c<%= comment.id%>uploading').hide()
-            $('#c<%= comment.id%>prompt').show()
-            var is_image = false
-            if (data.result['filename'].substr(-3,3) == "jpg") is_image = true
-            if (data.result['filename'].substr(-4,4) == "jpeg") is_image = true
-            if (data.result['filename'].substr(-3,3) == "png") is_image = true
-            if (data.result['filename'].substr(-3,3) == "gif") is_image = true
-            if (data.result['filename'].substr(-3,3) == "JPG") is_image = true
-            if (data.result['filename'].substr(-4,4) == "JPEG") is_image = true
-            if (data.result['filename'].substr(-3,3) == "PNG") is_image = true
-            if (data.result['filename'].substr(-3,3) == "GIF") is_image = true
```

Comment:
I kind of just deleted this code, but can definitely save some of it by either:
1. placing it back in the partial, but commenting it out
2. saving choice lines and moving it to `editorToolbar.js` (which used to be `dragdrop.js`)

Like, here's a similar thing happening in `editorToolbar.js`:
https://github.com/publiclab/plots2/blob/d2e249dcc9bc66f9b4523caeb9b345409bcff062/app/assets/javascripts/editorToolbar.js#L103-L107

Which JS is preferable in this instance?

Reply:
` if (['gif', 'GIF', 'jpeg', 'JPEG', 'jpg', 'JPG', 'png', 'PNG'].includes(extension)) `

this seems like a MUCH more compact way to do this. And, `editorToolbar.js` has the CSV detection too! Nice! Let's just rely on that more refined code! 👍

Reply:
Good to know! Finished resolving merge conflicts about an hour ago. I restored the code and left it commented out.